src/main/java/Parser.java

```diff
@@ -0,0 +1,249 @@
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Locale;
+
+public class Parser {
+    private ArrayList<String> validOperators;
+    private static final String timeFormat = "yyyy-M-d H:mm";
+
+    public Parser(){
```

Comment:
Perhaps there should be a space between "Parser()" and "{"? 🤔

Reply:
Thank you so much for patiently pointing out all my 'space' issues! I'll correct them asap:)